pkg/kp/pcstore/consul_store.go

```diff
@@ -44,9 +45,24 @@ func (s *consulStore) Create(
 	clusterName fields.ClusterName,
 	podSelector klabels.Selector,
 	annotations fields.Annotations,
+	session Session,
 ) (fields.PodCluster, error) {
 	id := fields.ID(uuid.New())
 
+	unlocker, err := s.lockForUpdateCreation(podID, availabilityZone, clusterName, session)
+	if err != nil {
+		return fields.PodCluster{}, err
+	}
+	defer unlocker.Unlock()
+
+	existing, err := s.FindWhereLabeled(podID, availabilityZone, clusterName)
+	if err != nil {
+		return fields.PodCluster{}, util.Errorf("Couldn't determine if pod cluster exists already: %v", err)
+	}
+	if len(existing) > 0 {
+		return existing[0], util.Errorf("Pod cluster already exists for %v", pcCreateLockPath(podID, availabilityZone, clusterName))
```

Comment:
i guess that forces you to deal with `len(existing)` which is annoying so feel free to ignore